test/integration/stats_integration_test.cc

```diff
@@ -54,6 +54,22 @@ TEST_P(StatsIntegrationTest, WithoutDefaultTagExtractors) {
   EXPECT_EQ(counter->tags().size(), 0);
 }
 
+TEST_P(StatsIntegrationTest, WithLargeRegex) {
+  config_helper_.addRuntimeOverride("re2.max_program_size.error_level", "1000");
+  config_helper_.addConfigModifier([&](envoy::config::bootstrap::v3::Bootstrap& bootstrap) -> void {
+    auto pattern = bootstrap.mutable_stats_config()
+                       ->mutable_stats_matcher()
+                       ->mutable_inclusion_list()
+                       ->add_patterns();
+    pattern->mutable_safe_regex()->mutable_google_re2();
+    pattern->mutable_safe_regex()->set_regex(
+        "cluster.(metadata-cluster|iam-cluster|service-control-cluster|backend-cluster-sample-"
+        "backend-s3fctubhaq-uc.a.run.app_443).upstream_rq_(reset|timeout|[1-5]xx)|listener.*");
```

Comment:
out of curiosity can you add a code comment indicating what program-size is for that?

I thought for this kind of test you might have set the error threshold much *smaller* so any regex would violate.

Reply:
This is kind of confusing.  Even with this fix:  re2.max_program_size.error_level=1000 is not applied.  It is still using default of 100.  This test only verified the program_size is not checked if runtime is not initialized. 

Since if it is checking, it will be checking with default max_program_size of 100, having a large program_size (>100) is the key to this test.   This program size is 180 .

Adding comment here is important.

Reply:
added comments.  Again this restores prior behavior where we bypass the checks.

Reply:
I think what you are saying is that the regex limit of 1000 set in the test does not matter, and neither does the actual size of the regex, because it is not being checked.

It's confusing still because the regex you added is large-ish but I'm skeptical it's got a cost of 1000, but really that doesn't matter. Maybe just add a comment there also?

Reply:
yeah it doesn't have a size of 1000, but the regex was over 100.  basically we'll fail prior to this change becasue the regex is checked. we'll also pass with a size of 10 with this change, but I dont' want to regression test because it's bad behavior I think we want to fix (as commented on the linked issue)
once I submit, I'll change the linked issue to "we should actually check the configured limit once runtime loads" but I don't think I'm on the hook for that part so it'll be help wanted

Reply:
Sounds good -- if you could just add that comment the test-case better as it is confusing in its current state. Otherwise the same will have to be conveyed to whoever picks this up :)

Reply:
could you maybe tell me what comment you want me to add?  I've done 2-3 tweaks so far and I think I'm missing the information you want me to convey :-)

Reply:
Sorry -- I'm good. I missed one of the comments you already added.

Reply:
you can't "from envoy" - arguably if google wanted actual size checks and wanted them enforced we could set re2_max_program_size and runtime_initialized on the command line but it'd be asking for trouble :-P
I think we need a proper Envoy fix for the longer term issue.